File: zenscript-code-model/src/main/java/raylras/zen/model/type/ErrorType.java
package raylras.zen.model.type;

import raylras.zen.model.CompilationEnvironment;

public enum ErrorType implements Type {

    INSTANCE;

    @Override
    public String getTypeName() {
        return "ERROR";
    }

    @Override
    public boolean isSuperclassTo(Type type) {
        return false;
    }

    @Override
    public boolean isCastableTo(Type type, CompilationEnvironment env) {
        return false;
    }

    @Override
    public String toString() {
        return getTypeName();
    }

}

File: zenscript-code-model/src/main/kotlin/raylras/zen/model/type/ErrorType.kt
package raylras.zen.model.type

import raylras.zen.model.CompilationEnvironment

object ErrorType : Type {
    override val typeName: String = "ErrorType"

    override fun isSupertypeTo(type: Type) = false

    override fun isCastableTo(that: Type?, env: CompilationEnvironment) = false

    override fun toString(): String = typeName
}
